src/Agent.Worker/TaskManager.cs

```diff
@@ -308,6 +310,29 @@ private async Task DownloadAsync(IExecutionContext executionContext, Pipelines.T
             }
         }
 
+        private void CheckTaskDeprecation(IExecutionContext executionContext, Pipelines.TaskStepDefinitionReference task)
+        {
+            string taskJsonPath = Path.Combine(GetDirectory(task), "task.json");
+            string taskJsonText = File.ReadAllText(taskJsonPath);
+            JObject taskJson = JObject.Parse(taskJsonText);
+            var deprecated = taskJson["deprecated"];
+
+            if (deprecated != null && deprecated.Value<bool>())
+            {
+                int majorVersion = new Version(task.Version).Major;
+                string deprecationMessage = $"Task '{task.Name}' version {majorVersion} is deprecated";
+                var removalDate = taskJson["removalDate"];
+
+                if (removalDate != null)
+                {
+                    string removalDateString = removalDate.Value<DateTime>().ToString("MMMM d, yyyy");
+                    deprecationMessage += $" and will be removed on {removalDateString}";
```

Comment:
>and will be removed on [](http://example.com/codeflow?start=45&length=22)

I'd adjust to "may not be available after {removalDateString}. "  Also should contain link for more info e.g. aka.ms/ado/taskremoval

Reply:
Added the help URL for tasks which have the help URL `https://docs.microsoft.com/azure/devops/pipelines/tasks/<category>/<task-name>`